File: app/src/androidTest/java/dev/bee/kanjianki/data/ContentValuesBuilder.java
package dev.bee.kanjianki.data;

import android.content.ContentValues;
import android.database.sqlite.SQLiteDatabase;

final class ContentValuesBuilder {
    private final SQLiteDatabase db;
    private final String table;
    private final ContentValues values = new ContentValues();

    private ContentValuesBuilder(SQLiteDatabase db, String table) {
        this.db = db;
        this.table = table;
    }

    static ContentValuesBuilder insert(SQLiteDatabase db, String table) {
        return new ContentValuesBuilder(db, table);
    }

    ContentValuesBuilder put(String key, String value) {
        values.put(key, value);
        return this;
    }

    ContentValuesBuilder put(String key, int value) {
        values.put(key, value);
        return this;
    }

    ContentValuesBuilder put(String key, long value) {
        values.put(key, value);
        return this;
    }

    ContentValuesBuilder put(String key, double value) {
        values.put(key, value);
        return this;
    }

    void commit() {
        db.insertOrThrow(table, null, values);
    }
}

File: app/src/androidTest/java/dev/bee/kanjianki/data/ContentValuesBuilder.kt
package dev.bee.kanjianki.data

import android.content.ContentValues
import android.database.sqlite.SQLiteDatabase

internal class ContentValuesBuilder private constructor(
    private val db: SQLiteDatabase,
    private val table: String,
) {
    private val values = ContentValues()

    fun put(key: String, value: String): ContentValuesBuilder = apply {
        values.put(key, value)
    }

    fun put(key: String, value: Int): ContentValuesBuilder = apply {
        values.put(key, value)
    }

    fun put(key: String, value: Long): ContentValuesBuilder = apply {
        values.put(key, value)
    }

    fun put(key: String, value: Double): ContentValuesBuilder = apply {
        values.put(key, value)
    }

    fun commit() {
        db.insertOrThrow(table, null, values)
    }

    companion object {
        @JvmStatic
        fun insert(db: SQLiteDatabase, table: String): ContentValuesBuilder =
            ContentValuesBuilder(db, table)
    }
}
